src/kernel/chainparams.cpp

```diff
@@ -537,10 +537,10 @@ class CRegTestParams : public CChainParams
         consensus.SegwitHeight = 0; // Always active unless overridden
         consensus.MinBIP9WarningHeight = 0;
         consensus.powLimit = uint256{"7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"};
-        consensus.nPowTargetTimespan = 14 * 24 * 60 * 60; // two weeks
+        consensus.nPowTargetTimespan = 24 * 60 * 60; // one day
```

Comment:
> In order for the test to run faster, we reduce its difficulty retarget period to 144,

The difference between using 144 and 2016 = mining_basic.py takes 30sec longer on my machine, which doesn't seem like such a big deal that we should change the difficulty adjustment period. I don't feel strongly, but it could be easier to just count mining_basic.py as one of the 30-60sec functional tests. :shrug:

Reply:
It felt a lot longer on my 2019 MBP, and I'm a bit worried the valgrind / sanitizer CI would suffer. But haven't done benchmarking.

I think we should have changed this to 144 back when `nMinerConfirmationWindow` was introduced in #7575. Not sure why it wasn't, @sipa?

Reply:
That makes sense.

I don't think it's a big deal though. There's two places where `nPowTargetTimespan` is accessed outside of test code:

1. In `CalculateNextWorkRequired` after the early return in regtest due to `fPowNoRetargeting`
2. In `PermittedDifficultyTransition` used by `headerssync.{h,cpp}`, after the early return in regtest due to `fPowAllowMinDifficultyBlocks`
3. In the value of `DifficultyAdjustmentInterval()` which is used:
  i. in `ContextualCheckBlockHeader()` for the timewarp attack, which we're testing here
  ii. in `miner.cpp` which we're testing here
  iii. in the `getnetworkhashps` RPC call, which we adjust the test for
  iv. in `GetNextWorkRequired` which on regtest always returns `params.powLimit`

Reply:
With debug enabled, 2016 blocks takes 3.4 minutes for me, vs 20 seconds for 144. Without debug, it's 54 seconds for 2016 and 15 seconds for 144.

Reply:
If we want to simulate a full timewarp attack, we'd need multiple such periods too.

Reply:
I'm convinced, marking resolved.